antarest/study/storage/generic_storage_service.py

```diff
@@ -0,0 +1,493 @@
+                        settings=settings,
+                        completionDate="",
+                        referenceStatus=(reference == output),
+                        synchronized=False,
+                        status="",
+                    )
+                )
+        return results
+
+    def import_output(
+        self,
+        metadata: T,
+        output: Union[IO[bytes], Path],
+    ) -> Optional[str]:
+        """
+        Import additional output on a existing study
+        Args:
+            metadata: study
+            output: new output (path or zipped data)
+
+        Returns: output id
+        """
+        path_output = (
+            Path(metadata.path) / "output" / f"imported_output_{str(uuid4())}"
+        )
+        os.makedirs(path_output)
+        output_name: Optional[str] = None
+        try:
+            if isinstance(output, Path):
+                if output != path_output:
+                    shutil.copytree(output, path_output / "imported")
+            else:
+                extract_zip(output, path_output)
+
+            fix_study_root(path_output)
+
+            ini_reader = IniReader()
+            info_antares_output = ini_reader.read(
+                path_output / "info.antares-output"
+            )["general"]
+
+            date = datetime.fromtimestamp(
+                int(info_antares_output["timestamp"])
+            ).strftime("%Y%m%d-%H%M")
+
+            mode = "eco" if info_antares_output["mode"] == "Economy" else "adq"
+            name = (
+                f"-{info_antares_output['name']}"
+                if info_antares_output["name"]
+                else ""
+            )
+
+            output_name = f"{date}{mode}{name}"
+            path_output.rename(Path(path_output.parent, output_name))
+
+            data = self.get(
+                metadata,
+                f"output/{output_name}",
+                -1,
+            )
+
+            if data is None:
+                self.delete_output(metadata, "imported_output")
+                raise BadOutputError("The output provided is not conform.")
+
+        except Exception as e:
+            logger.error("Failed to import output", exc_info=e)
+            shutil.rmtree(path_output)
+
+        return output_name
+
+    def export_study(
+        self, metadata: T, target: Path, outputs: bool = True
+    ) -> Path:
+        """
+        Export and compresse study inside zip
+        Args:
+            metadata: study
+            target: path of the file to export to
+            outputs: ask to integrated output folder inside exportation
+
+        Returns: zip file with study files compressed inside
+
+        """
+        path_study = Path(metadata.path)
+        return self.export_file(path_study, target, outputs)
+
+    def export_study_flat(
+        self, metadata: T, dest: Path, outputs: bool = True
+    ) -> None:
+        path_study = Path(metadata.path)
+        self.export_flat(path_study, dest, outputs)
```

Comment:
make this abstract
maybe some of the content of the variant and raw implementation can be shared in some static utility function